### DISCUSSION TITLE
Certificates not generating.

### DISCUSSION THREAD

**Discussion by Venkatesh-Dayani23** [Sunbird-Ed/Community#617] [Bugs] [answered]
Hi Team,
We are currently using Sunbird 5.0.0. Certificates are not getting generated even after the course is consumed 100% we are getting this error logs in certificate-generator flink job.Can someone guide us how to fix this ?

Certificate data: JobRequest {map={actor=Map(id -> Certificate Generator, type -> System), eid=BE_JOB_REQUEST, edata=Map(name -> Government of Tekdi, tag -> 0138737119411732482, criteria -> Map(narrative -> Gove
rnment of Tekdi), issuedDate -> 2023-09-01, data -> List(Map(recipientName -> Tekdi Dhananjay Test, recipientId -> f64268b3-2fcc-4fac-aefa-a11bf4b0cad2)), issuer -> Map(name -> Tekdi sachin test sandbox, url -> 
https://org_site.org), orgId -> 0137236500887961602, basePath -> https://dev.sunbirdsaas.com/certs, signatoryList -> List(Map(name -> CEO, image -> data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHAAAAAuCAYAAADwZ
J3MAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAABmJLR0QA9QD1AAoqTTx3AAAAB3RJTUUH5AsSBSIel7fkmQAADpNJREFUeNrtnHl4lPW1xz+zZ89ksidkc7KRPZmEEBISIYhsIpFN3GotUaxWfVr76HNvW++1ep/H
pbWKXazVohQhIksFQQIESEgCSchCFrLv+zpkm8nMZN77h7e1fa4rkEyCfv6cd2bec97v/H7n/M45z4gmJmwEvoOYBSlmswipxGhpU64LqaUNmF0kaMdU1DSEkZ2Xiounnu0Zb2ElG7W0YdfMTS+g3mBHb78PNU0RFFdEcbEikP5BV/z9O7g75gwSscnSJl4XoptvCxUxNulAc0cgJVU
xlFyOpqo1nJZ2H8zjY6xekce2u04Qv7AEpd0gML/dvylWoMlsxcCQB7XNwVyqjqWm0ZchrRNK+0kmZd4MSjxQBzSSeeduNtx2DDubq5Y2+YYxb1eg2Syjb8iLsisazhYl0Nbpi8g8QGhIKzGh9djbSckrXUJxbSyJ4UVsW7WXIN9awGxp028o80xACYMjbpRUasgtSeRKUxASDMRHl5
K+pJAgvzpsrSc5U7SSN/c9gpt7F5kZe9CEFCGVTFna+BlhXghoNFnT2BFKzqXVnMpLZEJnYFF4GcsWXyAyqBpX1QAizPQO+bPr+L1kF93KxtTD3HN7Fk72g5Y2f0aZ0wLqphwor43ncM5KTp1PYcLkyKP372F98j583duRiD/bDs2ClNzSFbyR9RRyex071v+B1KgcxKJpS7tw3QjIE
WEGvjhbnpNJjHbMk/yypRw6lUpXrzOLNZeRysDPtY0HVv0FJ7vPV1XPoB+7j97LqbJklifm89Ca93FTtqM3WGEyyrGzHWP+xT0Jw1ddqGqIJOfScsJDr5CRnIVUavh/75wTAhpNVgyMuKPTySkui+LQydvpGV1AUkIRTzzwR7Qjbvxt70aWJ1bgYKMFQBDEFFcn88YHjyNIdTyX+TKx
geW0dXizJyuT4mI3DFPWbNpYyqY7PkEq0Vvaza9GkDB01Yn61nByChK5ULeEEcEZD6dugoPrMAuiL/yYRQUUBCkVtXHsOZxB4aUYtH16enulJCzpYOcv/oNwdQ06vQ2Zv3mCiXETafHnkYhNTOrt2PfpZvac2sTKhHweWLeHjj4ffv7bFym66INZN46nqonBUWfe2HUPSYkV+Hq00to
VQHnNQkyGaVamFeBoP2JJ9xEECQMjXtR0x5FzKZ7KnnBG2gVcRR0kagpJXVRCpH8FjrbDiEVfvItYTECD0Yas49t499CdCCI5tVeU+KiquH3VJM09gTS3qokMqkQ75kJDdyRB6jY04aX0D7rx6gc/o6QuhCe2vk3ELc38cf9DZJ9fSoBvM0//+D3iQsrwdBti18ePcjg7gLEJR97d/w
B5JfFEhLRi0I9zq+GiRfw2CwoGtK7UtoVwpmoJpa1JGEUOBLi2cPuiQpLuPsctLtXY2U4iEn19emIRAXVTjryZ9RRHcpfww81ZhHlXsv3RTDbdmc1jO3LZe/wBXnv/YXy92pgWZIxOWHPX2mI6BtW8susRRCIFLz35OqNaJY+9+jsk1np+ev9OVibnorQbAsQcz9/K0YJ0tm7M5sAna
+kecuPJ7buJCryEwSjDSj45a/5OTyvoGvSlojmaosbFNPYFMSXIWODdwT3pHxHnV4W/2xXk0nG+bWVo1gUUBAn7j23h+PnFPP/YS6TFnMNkEvPj7YfZty8I7chi1m/IQ60O59DJLSRFFyK30mEQufHs7/+b8MA6fnb3CzS3h/Crt58hJT6PJza/hbeqEYCrY84cPrmKl3Y9jvcCHQXZ
anIvqrn/wfO4OfYiERuxVhiu04tv4CdSeocWUNYUS37DrdQOBCMxTBN7SxmZq94jyKsGd8dO5JLx67rPrB8jOvpCeer559my7kO2rD74fykyGIxydu+N552/JuLp54R/rDcll0NYFFrOnoMpKAOteHBdFg+v/QNSGWT+8jVkCj3/88TLmA1TVNSFc6kinPLKAHr67XF1GUehdOZCnhd
pSRVYO1mhmxTx0jMv4OPWOGP+jU/aU98VyYmy2yisXYRYgLiIWpZGnGahey1uDu2Ib2ABfZYFlPHnD3dw4nQAf37pBZyVw/92VRDEdPZ4Ut0UTVFDKh8eSGSwS2BySoVHsIKkiFJsZMMYsOf0mQic7HpQqwcwGcQ0DfnS32lHRupZMn/wIUF+TZwpTOfVnWt4+7XfolROsf2ZF8m8ez
d3ph+5oV6NTaqo7Y3kQpWGuvYghkyeuCmHSV2YT1LgGTxUHYhFM9N3nNUtdNosoazUm8jAqn+KZxakjIyq6OrzoKJRQ019EF0DAdQ3+NHdYMJokpOYWMqjDx5AbjWNeVrKiFZJT4sUiULPfRv2Ud+WgL7Cl42rsvnJpjdxduwFwDBxFVv5CCrlMErHCby8xhkbtbohvuj09rT0BlJQv
5SC6gRGzFZ4ug6TrikhPjAfL1UbMsnEjD/TWRVQIhawkk0wOS6jrVtNbauasyWp1LaGMGUS4aYaZHFYJaGB/TR2+2GrdmeiU4Sn+wh3LD+JXPZZ7GpsDeTI6VsJDWpnUKvmbF06P9ywlw0p+1BIx/55v6DgUUavOnLitIYVy+oZ6BEzFWEFiLiWNpKAlK4hP87XpZJdlkbXiBp3lx5S
I4pICzuGj1sLVjLdbD7S2dlCTdO2jI7b0dobwqvv/pSCU1LUMUYcVHrCAppIirpIkF87fu5NDGuVPPvGy4wYbLl37Sle2fkQ/T2jPHnfO2SsyaGsOobXdj/OgNmHcJ8q3Jz1/GBjFknBpxBhRBBEmMw2SMQGTNNyXvjNwxw6uQwvrykM43p2vvwyEYGXv5X9Y5MqqhqjOV13G0UtUUh
l02j8q1galk+UTzFKm4FZFW1WBBybdKa+LZyKKjUFxZH0an3RGe1xczcyOqhDPynjxWd3sjgqF/gsPvQOefGLN55jzOTLczt+hZdqgHue/RMd/d6MtEzgYtvMgNaHcRyRWMtx9jKyY9tZFthXMzSkoHPAA+O0DIPRHpV1C8sTzxAc2MmuI/fR32vDhvSTJMcVIhZ/fY3UZLKire8WCi
8nkducwsi4M4ELWkkKu0iCXx4eTp2IRJYv0c2YgMfy7+fFtx4GwxgLQxtJiG1hUWgBgb7t6PQy/npgG5fro0nUFBEe0IzSuo/3Dt3NkZIt/Gjrp7jYtVJYGc65nAAQRERFtGDUCVy+EkiIXzMqFwkKOwNIrJHJtLgpJ/Fw78fDpRcvVRcyYRAPVStq/15AhIAYEV+d/QmCjO6hAEobo
sitXUJ9/0IW2LeRHH2RxaHl+DpdQSqeWyW5GRNwZNSdpk4/VA5avD16/i02ARhNthRVL+FoYTotnf7UV3pzdciK4MgOPH2msFf0Ibc2UFkVzNVBe5ZqSqjrUrNmWT4bbj2ErfUUYtEUZkGGVDKBXGZEjIFriW3jOhVV7dHkVq+grFGDSGpAE3KBlLDzRHhVYKuYux18i7eTzIKMhs5I
Hnnxd6xJ/oTMDW9jY6VHIZtkbMKZF17/Obv2LGVxShtP73ibJdH5SMTXfxA3m+W09wVQeCWRs1W3MzDlhCa0gmVBpwldUIPKrg+Y++0oi3cjxCIjVnIdZmEaB6tepqbkaLX2lNfG8PfT6zh/Pgh3by3/9fTviQ0q4PqGkERox92paFpEdsVS6gaCcHboJSWukJTgM/i6NNyQH8dsYvE
VCGCYduD1vf/JwaPh2FhPgVjA1kZHUkw1E5NOFJeH8bdXM3F16r2m79cbHKntjORcVTKVXRr0RjlxAaWkhOey0LMUe+v5Oxc6JwQE0BmcqW4JY2DQGqWTEbVXA27KPl55/3ma2r14/dnHUEi/ed3QLMjpHvDnXPkKzpYlozXao/bpIC2umFi/XNwd2r82qZkPWHwL/QfW8iHiQ/Ig5P
PXzIKU4RF7An0bUUi/WfdgcsqJyuZojl7YQEmNGqWTkWUxOaTHnsPXtRmZdOarI7PJnBHwizCZpGi7BokLquerxyJEjIx5UnA5lY8LbqN9yJvggDYe3byXlLCzqOy6Le3KjDGnBRSJRISqm/DzaP3C62ZBQX1XFCdLUiksX4RIomBR2CUe3/IOoQvKkUlmr+dnsWc0V2Lgl6GfskcqM
SCVfj7XOT2toKYjnkP5d5FbrcFFNczmxCxSo3Nxdehj/g0xXTtzXsB/ZVzvzuUGDcfyU7nUnYSXWw8Z8ftJjsjDya7f0uZZhHkgoIje4QBOlqwmpyQFvcmOxMhyksNPEe5fgY1ca2kDLcqcjYGCIKOzz49jBbfxaekaBImU5ZpC1icdwt/1CtwER4AbgcUE1BlUdI1q8FVeQP4vdVKT
2YYrLSF8kruGc1XLkcinWLv8DOtjP8LbpZN/dC6+5zMsMNQk5UprHKfL15AQXsYtzp81QA1GGyoa4vm4aCMX66Jwsevh3jsOsCL6BB5OLXyXEpNvw6zGwKHRBRw8mUFFcwzrl/2d9LhPMZrsKK1LZP+nayhviyY6tpl1i7NJCDiFo813MzH5NsyKgNNmK84Urebg+Qz83JvZuiILF+U
wF6vTOJS7msruKKIDytma9hGa0EKspFpLP5d5wwwLKKazL5R9J7eQV5TAvXd9xIqEXEprE9hzNIPGAV+Wai6xPuUIMQEXvxfuGpgxAXUGJScKN3HwxCpsHXRsW3uc8XEpB3LuoGXIF01oKXelHSE+KA+ZeOz6b/gdZUYE7OwL4S/7t1PXoWbz2hNYK8b5OG8VjZ3upEZeZF1aNhEBpb
Mydnezc0MFNJmtyS1dx873tmGQu3FnWh61jZ5UtniRGFnDtpV7WOhXgVg0v5qmc5kbJuDVcVd2HcrkT1nbGJt2wMtjHIVskPjwKrat3EdccAlSyezOTH4X+NJzoCBImBaskIon+boxhrFJFb9+65e8/8FazAozfv5trEjMZ33qSWKDilHIvo9xM8WXH+RFwjee8tIbbNALEtZm5JGiu
UByZDGBXtVzbgTvZuSGbKGCIGZCb4dYLMJGMcp8//ej+cT/Av+gEo24J1rBAAAAJXRFWHRkYXRlOmNyZWF0ZQAyMDIwLTExLTE4VDA1OjM0OjMwLTA1OjAwqILoRAAAACV0RVh0ZGF0ZTptb2RpZnkAMjAyMC0xMS0xOFQwNTozNDozMC0wNTowMNnfUPgAAAAASUVORK5CYII=, de
signation -> CEO, id -> CEO/CEO)), svgTemplate -> https://sunbirdsaasdev.blob.core.windows.net/content/content/do_113844021840846848150/artifact/do_113844021840846848150_1689944072620_certificate_2023-07-21_18_2
4.svg, templateId -> do_113844021840846848150, related -> Map(batchId -> 0138737119411732482, courseId -> do_113873704209850368136, type -> Government of Tekdi), oldId -> , userId -> f64268b3-2fcc-4fac-aefa-a11b
f4b0cad2, courseName -> Tekdi test normal course  sachin 109), ets=1693568684931, context=Map(pdata -> Map(ver -> 1.0, id -> org.sunbird.learning.platform)), mid=LMS.31464fc9-80c3-46a6-ae91-a7f157f67255, object=
Map(id -> f64268b3-2fcc-4fac-aefa-a11bf4b0cad2, type -> GenerateCertificate)}}                                                                                                                                     
2023-09-06 10:53:57,017 INFO  org.sunbird.job.certgen.functions.CertificateGeneratorFunction [] - Certificate generator | is rc integration enabled: true                                                          
2023-09-06 10:53:57,030 INFO  org.sunbird.job.certgen.functions.CertificateGeneratorFunction [] - Certificate rc called | Api:: service.rc.search.api                                                              
2023-09-06 10:53:57,069 ERROR org.sunbird.job.exception.InvalidEventException              [] - Error while processing message for Partition: 0 and Offset: 7. Error : head of empty list                          
java.util.NoSuchElementException: head of empty list                                                                                                                                                               
    at scala.collection.immutable.Nil$.head(List.scala:430) ~[activity-aggregate-updater-1.0.0.jar:?]                                                                                                              
    at scala.collection.immutable.Nil$.head(List.scala:427) ~[activity-aggregate-updater-1.0.0.jar:?]                                                                                                              
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.callCertificateRc(CertificateGeneratorFunction.scala:244) ~[collection-certificate-generator-1.0.0.jar:?]                                    
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.generateRequest(CertificateGeneratorFunction.scala:205) ~[collection-certificate-generator-1.0.0.jar:?]                                      
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.$anonfun$generateCertificateUsingRC$1(CertificateGeneratorFunction.scala:142) ~[collection-certificate-generator-1.0.0.jar:?]                
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.$anonfun$generateCertificateUsingRC$1$adapted(CertificateGeneratorFunction.scala:125) ~[collection-certificate-generator-1.0.0.jar:?]        
    at scala.collection.immutable.List.foreach(List.scala:392) ~[activity-aggregate-updater-1.0.0.jar:?]                                                                                                           
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.generateCertificateUsingRC(CertificateGeneratorFunction.scala:125) ~[collection-certificate-generator-1.0.0.jar:?]                           
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.processElement(CertificateGeneratorFunction.scala:73) ~[collection-certificate-generator-1.0.0.jar:?]                                        
    at org.sunbird.job.certgen.functions.CertificateGeneratorFunction.processElement(CertificateGeneratorFunction.scala:33) ~[collection-certificate-generator-1.0.0.jar:?]                                        
    at org.sunbird.job.BaseProcessKeyedFunction.processElement(BaseProcessFunction.scala:113) ~[activity-aggregate-updater-1.0.0.jar:?]                                                                            
    at org.apache.flink.streaming.api.operators.KeyedProcessOperator.processElement(KeyedProcessOperator.java:83) ~[notification-job-1.0.0.jar:?]

**Comment by Krishnaj20** [comment]
@rajeevsathish - pls look into this.

**Comment by amitpriyadarshi** [comment]
@Venkatesh-Dayani23 As per the log, in class `CertificateGeneratorFunction` in method `callCertificateRc` line number 244 there is an exception. Post fetching the response from below registry search request, it is try to get the first element from the list and fetching `osId`. But it seems response is coming empty. Please validate why below search api is returning empty response.



```
curl --location '{{service.rc.basePath}}/PublicKey/search' \
--header 'Content-Type: application/json' \
--data '{
    "filters":{}
}'
```

Refer `collection-certificate-generator` flink job config file for the `service.rc.basePath` value.

cc: @Krishnaj20 @vinukumar-vs

**Reply by Venkatesh-Dayani23** [reply]
ok @amitpriyadarshi

**Reply by Venkatesh-Dayani23** [reply]
@amitpriyadarshi As per the details given when we execute the above curl we are getting empty value and we observed the below logs in the Registry service when we executed the curl. We are attaching the configmaps of registry service below.Kindly what to do further.

registry-5f8b6f6f47-tf9pc 11:00:32.839 [http-nio-8081-exec-4] WARN  d.s.registry.service.ISearchService:66 - Populates SearchQuery for limit/offset(optional): null                                                
registry-5f8b6f6f47-tf9pc 11:00:32.839 [http-nio-8081-exec-4] INFO  d.s.elastic.ElasticServiceImpl:110 - connection info: index:trainingcertificate connectioninfo:10.2.1.5:9200                                   
registry-5f8b6f6f47-tf9pc 11:00:32.840 [http-nio-8081-exec-4] INFO  d.s.elastic.ElasticServiceImpl:114 - resthighclient obj:org.elasticsearch.client.RestHighLevelClient@7185e611                                  
registry-5f8b6f6f47-tf9pc 11:00:32.844 [http-nio-8081-exec-4] ERROR dev.sunbirdrc.pojos.RequestWrapper:39 - Can't read from http stream. Set body empty                                                            
registry-5f8b6f6f47-tf9pc 11:00:32.844 [http-nio-8081-exec-4] ERROR dev.sunbirdrc.pojos.APIMessage:44 - Can't read request body                                                                                    
registry-5f8b6f6f47-tf9pc 11:16:35.294 [http-nio-8081-exec-5] WARN  d.s.registry.service.ISearchService:66 - Populates SearchQuery for limit/offset(optional): null                                                
registry-5f8b6f6f47-tf9pc 11:16:35.295 [http-nio-8081-exec-5] INFO  d.s.elastic.ElasticServiceImpl:110 - connection info: index:publickey connectioninfo:10.2.1.5:9200                                             
registry-5f8b6f6f47-tf9pc 11:16:35.295 [http-nio-8081-exec-5] INFO  d.s.elastic.ElasticServiceImpl:114 - resthighclient obj:org.elasticsearch.client.RestHighLevelClient@66f2ec1c                                  
registry-5f8b6f6f47-tf9pc 11:16:35.300 [http-nio-8081-exec-5] ERROR dev.sunbirdrc.pojos.RequestWrapper:39 - Can't read from http stream. Set body empty                                                            
registry-5f8b6f6f47-tf9pc 11:16:35.300 [http-nio-8081-exec-5] ERROR dev.sunbirdrc.pojos.APIMessage:44 - Can't read request body
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
CONFIG MAP:registry-schemas

Name:         registry-schemas
Namespace:    dev
Labels:       <none>
Annotations:  <none>

Data
====
TrainingCertificate.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "TrainingCertificate": {
      "$ref": "#/definitions/TrainingCertificate"
    }
  },
  "required": [
    "TrainingCertificate"
  ],
  "title": "TrainingCertificate",
  "definitions": {
    "TrainingCertificate": {
      "$id": "#/properties/TrainingCertificate",
      "type": "object",
      "title": "The TrainingCertificate Schema",
      "required": [
        "recipient",
        "training"
      ],
      "properties": {

        "status": {
          "type": "string",
          "enum": ["ACTIVE", "REVOKED", "DELETED"]
        },
        "jsonUrl": {
          "type": "string"
        },
        "recipient":{
          "$id": "#/properties/recipient",
          "$ref": "Recipient.json#/definitions/Recipient"
        },
        "templateUrl": {
          "type": "string"
        },
        "training":{
          "$id": "#/properties/training",
          "$ref": "Training.json#/definitions/Training"
        },
        "issuer":{
          "$id": "#/properties/issuer",
          "$ref": "Issuer.json#/definitions/Issuer"
        },
        "signatory":{
          "$id": "#/properties/signatory",
          "type":"array",
          "items":{
            "$ref": "Signatory.json#/definitions/Signatory"
          }

        },
        "certificateLabel":{
          "type": "string"
        }

      }
    }
  },
  "_osConfig": {
    "uniqueIndexFields": [
    ],
    "ownershipAttributes": [],
    "roles": [
    ],
    "inviteRoles": [
      "anonymous"
    ],
    "systemFields": ["osCreatedAt", "osUpdatedAt", "osCreatedBy", "osUpdatedBy"],
    "enableLogin": false,
    "credentialTemplate": "https://sunbirdsaasdev.blob.core.windows.net/content/schema/credential_template.json"
  }
}

Issuer.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "Issuer": {
      "$ref": "#/definitions/Issuer"
    }
  },
  "required": [
    "Issuer"
  ],
  "title": "Issuer",
  "definitions": {
    "Issuer": {
      "$id": "#/properties/Issuer",
      "type": "object",
      "title": "The Issuer Schema",
      "required": [
        "name"
      ],
      "properties": {
        "name": {
          "type": "string"
        },
        "url": {
          "type": "string"
        },
        "kid": {
          "type": "string"
        }

      }
    }
  }

}
PublicKey.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "PublicKey": {
      "$ref": "#/definitions/PublicKey"
    }
  },
  "required": [
    "PublicKey"
  ],
  "title": "PublicKey",
  "definitions": {
    "PublicKey": {
      "$id": "#/properties/PublicKey",
      "type": "object",
      "title": "The PublicKey Schema",
      "required": [
        "value"
      ],
      "properties": {
        "value": {
          "type": "string"
        },
        "alg": {
          "type": "string"
        }
      }
    }
  },
  "_osConfig": {
    "uniqueIndexFields": [
      "value"
    ],
    "ownershipAttributes": [],
    "roles": [
    ],
    "inviteRoles": [
      "anonymous"
    ],
    "systemFields": ["osCreatedAt", "osUpdatedAt", "osCreatedBy", "osUpdatedBy"],
    "enableLogin": false
  }
}
Recipient.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "Recipient": {
      "$ref": "#/definitions/Recipient"
    }
  },
  "required": [
    "Recipient"
  ],
  "title": "Recipient",
  "definitions": {
    "Recipient": {
      "$id": "#/properties/Recipient",
      "type": "object",
      "title": "The Recipient Schema",
      "required": [
        "id"
      ],
      "properties": {
        "id": {
          "type": "string"
        },
        "type": {
          "type": "string"
        },
        "name": {
          "type": "string"
        }

      }
    }
  }

}
Signatory.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "Signatory": {
      "$ref": "#/definitions/Signatory"
    }
  },
  "required": [
    "Signatory"
  ],
  "title": "Signatory",
  "definitions": {
    "Signatory": {
      "$id": "#/properties/Signatory",
      "type": "object",
      "title": "The Signatory Schema",
      "required": [

      ],
      "properties": {
        "identity": {
          "type": "string"
        },
        "designation": {
          "type": "string"
        },
        "image": {
          "type": "string"
        },
        "name":{
          "type":"string"
        }
      }
    }
  }

}
Training.json:
----
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "properties": {
    "Training": {
      "$ref": "#/definitions/Training"
    }
  },
  "required": [
    "Training"
  ],
  "title": "Training",
  "definitions": {
    "Training": {
      "$id": "#/properties/Training",
      "type": "object",
      "title": "The Training Schema",
      "required": [
        "id",
        "type"
      ],
      "properties": {
        "id": {
          "type": "string"
        },
        "batchId": {
          "type": "string"
        },
        "type": {
          "type": "string"
        },
        "name":{
          "type":"string"
        }
      }
    }
  }

}

BinaryData
====

Events:  <none>


Configmap: registry-config
-----------------------------------------------
Name:         registry-config
Namespace:    dev
Labels:       <none>
Annotations:  <none>

Data
====
elastic_search_enabled:
----
true
search_provider:
----
dev.sunbirdrc.registry.service.ElasticSearchService
workflow.enable:
----
false
elastic_search_connection_url:
----
XXXXXXXXXXXXXXXXXXXXXXX
template_base_url:
----
http://registry-service:8081/api/v1/templates/
connectionInfo_password:
----
XXXXXXXXXXX
connectionInfo_uri:
----
XXXXXXXXXXXXXXXXXXXXXX
connectionInfo_username:
----
XXXXXXXXXXXX
auditTaskExecutor_queueCapacity:
----
100
enable_external_templates:
----
true
sign_url:
----
http://certificatesign-service:8079/sign
signature_enabled:
----
true
taskExecutor_index_queueCapacity:
----
100
connectionInfo_maxPoolSize:
----
200
pdf_url:
----
http://certificateapi-service:8078/api/v1/certificatePDF
search_providerName:
----
dev.sunbirdrc.registry.service.ElasticSearchService

BinaryData
====

Events:  <none>

**Comment by ashokreddy1208** [comment]
@amitpriyadarshi @AmiableAnil Any update on this issue?

**Comment by amitpriyadarshi** [comment, marked as answer]
@Venkatesh-Dayani23 As per above stack-trace and analysis, the job is failing because of un-availability of public key in RC. Please create the public key and add it into RC. 
cc: @kumarks1122, @Krishnaj20 @vinukumar-vs

**Reply by AmiableAnil** [reply]
@Venkatesh-Dayani23 
Please use the below command to generate the public key

`openssl genrsa -out key.pem; cat key.pem; 
 openssl rsa -in key.pem -pubout -out pubkey.pem;
 cat pubkey.pem; rm key.pem pubkey.pem`

**Reply by Venkatesh-Dayani23** [reply]
ok @AmiableAnil

**Reply by Venkatesh-Dayani23** [reply]
Hi @AmiableAnil @amitpriyadarshi 
Could you please guide me on the process of adding public key to RC.

**Reply by amitpriyadarshi** [reply]
@Venkatesh-Dayani23 please schedule a call today adding me and Anil.

**Reply by Venkatesh-Dayani23** [reply]
Hi @amitpriyadarshi 
Can you kindly let me know at what time can we connect.

**Reply by amitpriyadarshi** [reply]
@Venkatesh-Dayani23 can we connect today at 1PM.
cc: @AmiableAnil @kumarks1122

**Comment by Venkatesh-Dayani23** [comment]
Hi @Amit ***@***.***>

Sure, we can have a call at 1:00 PM

Thanks & Regards,
Venkatesh Dayani
Sysops Engineer | Cloud Infrastructure Services
Opt IT Technologies India Pvt. Ltd
✉ : ***@***.***
✆ : +91 8919448467
________________________________

**Comment by Venkatesh-Dayani23** [comment]
Hi @Amit ***@***.***>

We figured out that few manual steps are missing we are working on it. So, we can cancel the call. Thank you so much for your help.

Thanks & Regards,
Venkatesh Dayani
Sysops Engineer | Cloud Infrastructure Services
Opt IT Technologies India Pvt. Ltd
✉ : ***@***.***
✆ : +91 8919448467
________________________________

**Reply by Krishnaj20** [reply]
@Venkatesh-Dayani23 - is your query answered?

**Comment by ashokreddy1208** [comment]
@Venkatesh-Dayani23 can you please get back incase your issue is not resolved. If its resolved, please mark the thread as answered

**Reply by Krishnaj20** [reply]
@gandham-santhosh - can you mark the thread as answered and close if there are no further questions from optit team?

**Reply by Krishnaj20** [reply]
@gandham-santhosh - following up to check if this thread is now answered?

**Reply by gandham-santhosh** [reply]
@Venkatesh-Dayani23  mention the steps that are followed here to generate the keys and mark it as answered if you are able to generate the certs
